packages/vats/test/bootstrapTests/test-vaults-upgrade.js

```diff
@@ -0,0 +1,228 @@
+// @ts-check
+/**
+ * @file Bootstrap test integration vaults with smart-wallet
+ */
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+
+import { Fail } from '@agoric/assert';
+import { Offers } from '@agoric/inter-protocol/src/clientSupport.js';
+import { makeAgoricNamesRemotesFromFakeStorage } from '../../tools/board-utils.js';
+import { makeSwingsetTestKit, makeWalletFactoryDriver } from './supports.js';
+
+/**
+ * @type {import('ava').TestFn<Awaited<ReturnType<typeof makeDefaultTestContext>>>}
+ */
+const test = anyTest;
+
+// presently all these tests use one collateral manager
+const collateralBrandKey = 'ATOM';
+
+const makeDefaultTestContext = async t => {
+  console.time('DefaultTestContext');
+  const swingsetTestKit = await makeSwingsetTestKit(t, 'bundles/vaults');
+
+  const { runUtils, storage } = swingsetTestKit;
+  console.timeLog('DefaultTestContext', 'swingsetTestKit');
+  const { EV } = runUtils;
+
+  // Wait for ATOM to make it into agoricNames
+  await EV.vat('bootstrap').consumeItem('vaultFactoryKit');
+  console.timeLog('DefaultTestContext', 'vaultFactoryKit');
+
+  // has to be late enough for agoricNames data to have been published
+  const agoricNamesRemotes = makeAgoricNamesRemotesFromFakeStorage(
+    swingsetTestKit.storage,
+  );
+  agoricNamesRemotes.brand.ATOM || Fail`ATOM missing from agoricNames`;
+  console.timeLog('DefaultTestContext', 'agoricNamesRemotes');
+
+  const walletFactoryDriver = await makeWalletFactoryDriver(
+    runUtils,
+    storage,
+    agoricNamesRemotes,
+  );
+  console.timeLog('DefaultTestContext', 'walletFactoryDriver');
+
+  console.timeEnd('DefaultTestContext');
+
+  const { readLatest } = swingsetTestKit;
+  const readRewardPoolBalance = () => {
+    return readLatest('published.vaultFactory.metrics').rewardPoolAllocation
+      .Minted?.value;
+  };
+  const readCollateralMetrics = vaultManagerIndex =>
+    readLatest(`published.vaultFactory.manager${vaultManagerIndex}.metrics`);
+
+  return {
+    ...swingsetTestKit,
+    agoricNamesRemotes,
+    readCollateralMetrics,
+    readRewardPoolBalance,
+    walletFactoryDriver,
+  };
+};
+
+test.before(async t => {
+  t.context = await makeDefaultTestContext(t);
+});
+test.after.always(t => t.context.shutdown());
+
+test.serial('open vault', async t => {
```

Comment:
does `test.serial` that they're run in the order in which they appear in the file?

Reply:
Correct! https://github.com/avajs/ava/blob/main/docs/01-writing-tests.md#running-tests-serially